### PR TITLE
Add support for OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION

### LINKED ISSUE #4153: Support for OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION
# Feature Request

Support
https://github.com/open-telemetry/opentelemetry-specification/blob/dd47c009da1de335c540decc5d5930e46ea61916/specification/metrics/sdk_exporters/otlp.md#additional-configuration

## Notes

Found while working on https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2009
The second missing env. var. is https://github.com/open-telemetry/opentelemetry-dotnet/issues/3756

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Fixes #4153

## Changes

This PR implements support for the `OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION` environment variable as specified in the [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/otlp.md#additional-configuration).

### Key Implementation Details

- **Internal API only**: All new types and properties are marked `internal` to avoid expanding the public API surface
- **Spec-compliant parsing**: Uses direct string comparison (case-insensitive) to validate against spec-defined snake_case values:
  - `explicit_bucket_histogram` (default)
  - `base2_exponential_bucket_histogram`
- **View precedence**: Explicit views configured via `AddView` take precedence over the default aggregation setting
- **Applied in both code paths**: Default aggregation is applied whether or not views are configured, ensuring consistent behavior

### Implementation Components

1. **MetricReaderHistogramAggregation** (internal enum): Represents the two valid histogram aggregation types
2. **MetricReaderOptions.DefaultHistogramAggregation** (internal property): Stores the configured default
3. **OtlpServiceCollectionExtensions**: Parses the environment variable/configuration and sets the property
4. **MetricReaderExt.CreateDefaultHistogramConfiguration**: Helper method that applies the default to histogram instruments
5. **Test coverage**: 11 new tests covering environment variables, IConfiguration, view overrides, and edge cases

### Testing

All existing tests pass (524/528), with 4 unrelated failures in other areas. New tests added:
- Configuration-based parsing
- Environment variable parsing  
- Application to histogram instruments
- View override behavior
- Invalid value handling

### Documentation

- Updated `README.md` with environment variable documentation
- Updated `CHANGELOG.md` with feature description

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry/Metrics/Reader/MetricReaderExt.cs [hunk 1]
```diff
@@ -52,7 +52,15 @@ internal virtual List<Metric> AddMetricWithNoViews(Instrument instrument)
...
         Debug.Assert(this.metrics != null, "this.metrics was null");
 
-        var metricStreamIdentity = new MetricStreamIdentity(instrument!, metricStreamConfiguration: null);
+        MetricStreamConfiguration? metricStreamConfiguration = null;
+
+        // Apply default histogram aggregation if configured
+        if (this.DefaultHistogramAggregation.HasValue)
+        {
+            metricStreamConfiguration = this.CreateDefaultHistogramConfiguration(instrument);
+        }
```
Comment (martincostello) [inline src/OpenTelemetry/Metrics/Reader/MetricReaderExt.cs:61]: This seems weird to me to just test for `HasValue` but not use it explicitly.

I think it would be better passed to the `CreateDefaultHistogramConfiguration()` method like:

```csharp
if (this.DefaultHistogramAggregation is {} aggregation)
{
    metricStreamConfiguration = this.CreateDefaultHistogramConfiguration(instrument, aggregation);
}
```

File: src/OpenTelemetry/Metrics/Reader/MetricReaderExt.cs [hunk 2]
```diff
@@ -207,6 +222,25 @@ private void CreateOrUpdateMetricStreamRegistration(in MetricStreamIdentity metr
...
+    {
+        Debug.Assert(instrument != null, "instrument was null");
+
+        var instrumentType = instrument!.GetType();
+        if (instrumentType.IsGenericType)
+        {
+            var genericType = instrumentType.GetGenericTypeDefinition();
+            if (genericType == typeof(Histogram<>))
+            {
+                return this.DefaultHistogramAggregation!.Value == MetricReaderHistogramAggregation.Base2ExponentialBucketHistogram
```
Comment (martincostello) [inline src/OpenTelemetry/Metrics/Reader/MetricReaderExt.cs]: Which also means we don't need this `!`.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -34,6 +34,23 @@ public static void AddOtlpExporterMetricsServices(this IServiceCollection servic
...
                 }
+
+                // Parse histogram aggregation using direct string comparison instead of Enum.TryParse.
+                // The spec defines snake_case values (explicit_bucket_histogram, base2_exponential_bucket_histogram).
+                // Using direct string comparison ensures we strictly validate against spec-defined values and fail
+                // gracefully for invalid inputs, rather than attempting to parse arbitrary strings to enum values.
+                var otlpDefaultHistogramAggregation = config[OtlpSpecConfigDefinitions.MetricsDefaultHistogramAggregationEnvVarName];
+                if (!string.IsNullOrWhiteSpace(otlpDefaultHistogramAggregation))
+                {
+                    if (otlpDefaultHistogramAggregation!.Equals("base2_exponential_bucket_histogram", StringComparison.OrdinalIgnoreCase))
```
Comment (martincostello) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpServiceCollectionExtensions.cs]: The `!` seems redundant.

Comment (martincostello) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpServiceCollectionExtensions.cs]: IMHO it's better to use the static `Equals` method that the instance one as it's inherently null-safe (and would make the `IsNullOrWhiteSpace` check redundant).

Comment (martincostello) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpServiceCollectionExtensions.cs]: Are we intentionally being forgiving on the case here, as the comment says we're being strict.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs [hunk 4]
```diff
@@ -971,6 +971,186 @@ public void Dispose()
...
+
+    [Theory]
+    [InlineData("explicit_bucket_histogram", MetricReaderHistogramAggregation.ExplicitBucketHistogram)]
+    [InlineData("base2_exponential_bucket_histogram", MetricReaderHistogramAggregation.Base2ExponentialBucketHistogram)]
+    [InlineData("invalid", null)]
+    internal void TestDefaultHistogramAggregationUsingEnvVar(string configValue, MetricReaderHistogramAggregation? expectedAggregation)
+    {
+        try
+        {
+            Environment.SetEnvironmentVariable(OtlpSpecConfigDefinitions.MetricsDefaultHistogramAggregationEnvVarName, configValue);
```
Comment (martincostello) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: [`EnvironmentVariableScope`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/10802913790a4b9bae19074d14618d90ec58a41e/test/OpenTelemetry.Tests/EnvironmentVariableScope.cs#L4) can be used to do this in a less-verbose manner than needing a `try-finally`.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs [hunk 5]
```diff
@@ -971,6 +971,186 @@ public void Dispose()
...
+            })
+            .Build();
+
+        var histogram = meter.CreateHistogram<long>("test_histogram");
+        histogram.Record(100);
+        histogram.Record(200);
+
+        meterProvider.ForceFlush();
+
+        var batch = new Batch<Metric>(exportedItems.ToArray(), exportedItems.Count);
```
Comment (martincostello) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: I'd move this into a using [like this test](https://github.com/open-telemetry/opentelemetry-dotnet/blob/10802913790a4b9bae19074d14618d90ec58a41e/test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/Implementation/Serializer/ProtobufOtlpMetricSerializerTests.cs#L59-L112) - there's been some flakiness where the export hasn't run on time as the instance hasn't been disposed and then the assertions fail.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs [hunk 6]
```diff
@@ -971,6 +971,186 @@ public void Dispose()
...
+        var request = CreateMetricExportRequest(batch, ResourceBuilder.CreateEmpty().Build());
+
+        var resourceMetric = request.ResourceMetrics.Single();
+        var scopeMetrics = resourceMetric.ScopeMetrics.Single();
+        var actual = scopeMetrics.Metrics.Single();
```
Comment (martincostello) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: Could use `Assert.Single()`.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpServiceCollectionExtensions.cs (hunk 3 above)
Comment (rajkumar-rangaraj) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpServiceCollectionExtensions.cs]: Updated the comment in the code

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests.csproj [hunk 7]
```diff
@@ -50,6 +50,7 @@
     <Compile Include="$(RepoRoot)\test\OpenTelemetry.Tests\Shared\TestActivityProcessor.cs" Link="Includes\TestActivityProcessor.cs" />
     <Compile Include="$(RepoRoot)\test\OpenTelemetry.Tests\Shared\TestEventListener.cs" Link="Includes\TestEventListener.cs" />
     <Compile Include="$(RepoRoot)\test\OpenTelemetry.Tests\Shared\Utils.cs" Link="Includes\Utils.cs" />
+    <Compile Include="$(RepoRoot)\test\OpenTelemetry.Tests\EnvironmentVariableScope.cs" Link="Includes\EnvironmentVariableScope.cs" />
```
Comment (martincostello) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests.csproj]: Nit: sorting

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs [hunk 8]
```diff
@@ -971,6 +971,159 @@ public void Dispose()
...
+            {
+                metricReaderOptions.DefaultHistogramAggregation = aggregation;
+            })
+            .Build();
+
+        var histogram = meter.CreateHistogram<long>("test_histogram");
+        histogram.Record(100);
+        histogram.Record(200);
+
+        meterProvider.ForceFlush();
```
Comment (martincostello) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: I'd wrap this in the `using` as per the previous comment and assert that the flush worked to try and avoid flakiness in this test and the two below.

Comment (rajkumar-rangaraj) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: ba30dfdb169799c97681a92f6b7d280e0b1449ce -I believe this is what you're looking for.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs [hunk 9]
```diff
@@ -971,6 +971,162 @@ public void Dispose()
...
+            Assert.Equal("test_histogram", metric.Name);
+
+            if (expectExponential)
+            {
+                Assert.Equal(MetricType.ExponentialHistogram, metric.MetricType);
+            }
+            else
+            {
+                Assert.Equal(MetricType.Histogram, metric.MetricType);
+            }
```
Comment (martincostello) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: I'd move these assertions outside of the `using`, then the disposal will have run first so we're sure it's definitely flushed everything out.

Comment (rajkumar-rangaraj) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: Fixed.
